### PR TITLE
Improve Box/Grid align styling

### LINKED ISSUE #5032: Box alignContent prop value doesn't map to proper css
### Expected Behavior

When using a `Box` component with alignContent prop set to between or around, etc. It should align content accordingly

### Actual Behavior

It injects invalid value to `align-content` property for that component.

### URL, screen shot, or Codepen exhibiting the issue
Using the following code we get the css:
```js
<Box direction="row" alignContent="between">
  <OtherComponent  />
  <OtherComponent  />
</Box>
```
image

### Steps to Reproduce

1. Use a `Box` component and set a value for the `alignContent` property.
2. Inspect the created element

### Your Environment

- Grommet version: 2.16.3
- Browser Name and version: Chrome 88.0.4324.150
- Operating System and version (desktop or mobile): OSX

### PR DESCRIPTION
**Pull request by leossantiago**
This PR fixes issue #5032 by mapping `alignContent` values properly, and also improves `align` and `alignContent` props usability on `<Box />` and `<Grid />` components.

#### What does this PR do?

 > `alignContent` mapping fix

`between` and `around` values are now mapped to `space-between` and `space-around`, becoming valid CSS values to `alignContent`;

> add more possibilities to `align` and `alignContent`
 
According to [THIS](https://developer.mozilla.org/pt-BR/docs/Web/CSS/align-content) and [THIS](https://developer.mozilla.org/pt-BR/docs/Web/CSS/align-items), `align` (which corresponds to `align-items`) and `alignContent` (which corresponds to `align-content`) can receive many values besides the ones supported by grommet. 

This PR proposes we accept the given values:

`align` : `['baseline', 'center', 'end', 'start', 'stretch']`

`alignContent` : `['around', 'baseline', 'between', 'center', 'evenly', 'end', 'start', 'stretch']`

Those are listed as possible values but, by adding `string` as a valid value as well, we allow our users to set any CSS valid value to those props, including composed ones such as `first baseline`, `last baseline`, `safe start` and `unsafe start`. Those will be considered if the user passes a value that doesn't match any of the given values above. 

> refactor align styling

Both properties are applied to Box and Grid and with the same possible values. That said, I considered that moving those to `utils/style.js` so we have less duplicated code, and to assure any changes will be reflected in both components. TS types were also moved, and are now imported as `AlignType`.

> etc

Docs and tests were updated to match the changes. Also removed doc default values of properties that default to undefined.

#### Where should the reviewer start?

`src/js/utils/styles.js`
`src/js/components/Box/StyledBox.js`
`src/js/components/Grid/StyledGrid.js`

#### What testing has been done on this PR?

Jest tests were updated, and manual testing.

#### How should this be manually tested?

You can play around with `align` and `alignContent` props using this component:

```js
const AlignBox = () => (
  <Grommet full theme={grommet}>
    <Box
      width="400px"
      height="200px"
      wrap
      background="brown"
      alignContent="between"
      margin="small"
    >
      <Box height="150px" background="gold">
        A
      </Box>
      <Box height="150px" background="silver">
        B
      </Box>
    </Box>
    <Box
      width="400px"
      height="200px"
      background="brown"
      align="stretch"
      margin="small"
    >
      <Box background="gold">BOX</Box>
    </Box>
  </Grommet>
);
```

#### What are the relevant issues?

Closes #5032 

#### Is this change backwards compatible or is it a breaking change?

Backwards compatible.

### CODE REVIEW

**Comment by leossantiago** [comment]
Hey @halocline ! I'm working on closing all my open pull requests, so since you assigned yourself to this one, feel free to ping me in case you need anything related to it :)

File: src/js/components/Box/__tests__/Box-test.js [hunk 1]
```diff
@@ -94,6 +94,9 @@ describe('Box', () => {
         <Box alignContent="around" />
         <Box alignContent="stretch" />
         <Box alignContent="end" />
+        <Box alignContent="around" />
```
Comment (halocline) [inline src/js/components/Box/__tests__/Box-test.js:97]: Duplicate of L94. Should be removed.
```suggestion
```

File: src/js/components/Box/__tests__/Box-test.js [hunk 2]
```diff
@@ -94,6 +94,9 @@ describe('Box', () => {
...
+        <Box alignContent="baseline" />
+        <Box alignContent="evenly" />
```
Comment (halocline) [inline src/js/components/Box/__tests__/Box-test.js:99]: If we are also going to support the caller passing valid CSS string values, should we also incorporate tests for those values as well (i.e. in addition to the mapped values)?

File: src/js/components/Grid/__tests__/Grid-test.js [hunk 3]
```diff
@@ -184,6 +185,9 @@ describe('Grid', () => {
         <Grid alignContent="around" />
         <Grid alignContent="end" />
         <Grid alignContent="stretch" />
+        <Grid alignContent="around" />
```
Comment (halocline) [inline src/js/components/Grid/__tests__/Grid-test.js:188]: Duplicate.

```suggestion
```

**Review by halocline — COMMENTED** [review]
Nice cleanups and enhancement to support valid CSS string values. Makes me want to see if there are other opportunities for similar mappings / value extensions.

Jest tests in Box and Grid have a duplicate test scenarios which should be removed. I am happy to remove if you agree.

Also, wondering if others agree that we should add snapshot tests for the "any valid CSS string values" since we will now be stating those are valid?

**Comment by halocline** [comment]
> Hey @halocline ! I'm working on closing all my open pull requests, so since you assigned yourself to this one, feel free to ping me in case you need anything related to it :)

We were just swarming PR reviews as a group to catch up.

Awesome. This looks great. Thank you for the cleanups. Dropped a couple minor comments and then this should be ready to go.

**Comment by halocline** [comment]
> @halocline totally missed your last messages. Just got an email with your pushes. Sorry about that.
> Think I'm - finally - on track to get back working on issues and PRs.
> By the way, whenever you feel the need, feel free to make changes to my PRs. The current and future ones :) as long as we keep helping grommet grow, there's no need to wait for me, so if that matters, you have my approval 😆
> 
> One question: after the changes, GH is pointing 173 files changed. Should we worry about that?

Hey @leossantiago! Yes, we should worry (kind of) 🤪 . They are a result from changes to [new Prettier defaults](https://prettier.io/docs/en/options.html#arrow-function-parentheses) which were applied while I was merging master into this branch. The changes are innocuous (formatting only), but we don't want them polluting this PR. I am going to apply the Prettier changes in a separate PR today so that this PR returns to its pristine state 😉  and is easy for folks to review.

Cheers!

**Comment by leossantiago** [comment]
@halocline yeah, I noticed they were just formatting changes, but I agree we should have this on a separate PR, merged before this one, so we can easily track the changes in this PR if needed. 😄

**Review by halocline — APPROVED** [review]
This PR is good for final review.

The TableCell/doc.js change is unrelated to this PR. It is a valid Prettier syntax change based on latest upgrade.

**Review by ericsoderberghp — APPROVED** [review]

**Pull request merged**